### PR TITLE
[WFCORE-371/WFCORE-419] : Expose mgmt client socket binding config through ModelControllerClient Factory.

### PR DESCRIPTION
**Pull request by ehsavoie**
Now you can specify the bind address for the ModelControllerClient.
You may set it using  client socket binding config through CLI connect with the --bind= param.

Jira: https://issues.jboss.org/browse/WFCORE-371 and https://issues.jboss.org/browse/WFCORE-419

### CODE REVIEW

**Comment by bstansberry** [comment]
Looks good. @aloubyansky, please approve or comment. Thanks!

File: cli/src/main/java/org/jboss/as/cli/impl/CliLauncher.java [hunk 1]
```diff
@@ -174,6 +175,12 @@ public static void main(String[] args) throws Exception {
                         argError = "'=' is missing after --timeout";
                         break;
                     }
+                } else if(arg.startsWith("--bind=") || arg.startsWith("bind=")) {
```
Comment (aloubyansky) [inline cli/src/main/java/org/jboss/as/cli/impl/CliLauncher.java]: Let's not support "bind=". That will be consistent with all the other supported arguments except for the controller.

**Comment by aloubyansky** [comment]
That's all I had. Otherwise, nicely done. Thanks!

File: cli/src/main/java/org/jboss/as/cli/impl/CommandContextConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        this.controller = controller;
+        this.username = username;
+        this.password = password;
+        this.clientBindAddress = clientBindAddress;
+        this.consoleInput = consoleInput;
+        this.consoleOutput = consoleOutput;
+        this.initConsole = initConsole;
+        this.disableLocalAuth = disableLocalAuth || username != null;
+        this.connectionTimeout = connectionTimeout;
+    }
```
Comment (ehsavoie) [inline cli/src/main/java/org/jboss/as/cli/impl/CommandContextConfiguration.java]: Done

**Pull request merged**